compiler/test/fail_compilation/var_func_attr.d

```diff
@@ -0,0 +1,35 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/var_func_attr.d(19): Error: cannot implicitly convert expression `__lambda8` of type `void function() nothrow @nogc @safe` to `void function() pure`
+---
+*/
+
+// Test the effect of function attributes on variables
+// See:
+//   https://issues.dlang.org/show_bug.cgi?id=7432
+//   https://github.com/dlang/dmd/pull/14199
+// Usually it's a no-op, but the attribute can apply to the function/delegate type of the variable
+// The current behavior is weird, so this is a test of the current behavior, not necessarily the desired behavior
+
+// No-op
+pure int x;
```

Comment:
If this is a no-op should it be allowed? It's quite confusing that D allows this spaghetti in the grammar

Reply:
Like the comment above says, this is only to put the existing behavior in the test suite, not to indicate what the 'correct' behavior is.